src/renderer/coremods/commands/index.ts

```diff
@@ -9,35 +11,135 @@ import { loadCommands, unloadCommands } from "./commands";
 const logger = Logger.api("Commands");
 const injector = new Injector();
 
-interface ApplicationCommandSearchStoreMod {
-  useDiscoveryState: (...args: unknown[]) =>
+interface ApplicationCommandIndexStore extends Store {
+  getContextState: (channel: Channel) =>
     | {
-        sectionDescriptors: RepluggedCommandSection[];
+        fetchState: { fetching: boolean };
+        result: {
+          sectionIdsByBotId: Record<string, string>;
+          sections: Record<
+            string,
+            { commands: Record<string, AnyRepluggedCommand>; descriptor: RepluggedCommandSection }
+          >;
+          version: string;
+        };
+        serverVersion: symbol | string;
```

Comment:
Can you make a type just for this? This exact record is copied and pasted for almost every function below, resulting in duplicate code.